datafusion/core/src/execution/session_state.rs

```diff
@@ -74,10 +74,23 @@ use std::fmt::Debug;
 use std::sync::Arc;
 use uuid::Uuid;
 
-/// Execution context for registering data sources and executing queries.
-/// See [`SessionContext`] for a higher level API.
+/// `SessionState` contains all the necessary state to plan and execute queries,
+/// such as configuration, functions, and runtime environment. Please see the
+/// documentation on [`SessionContext`] for more information.
```

Comment:
I wanted to make it clear that SessionContext not only has a higher level API it also has a good description of the differences between the structures:

https://docs.rs/datafusion/latest/datafusion/execution/context/struct.SessionContext.html#sessioncontext-sessionstate-and-taskcontext